superset-frontend/src/views/CRUD/hooks.ts

```diff
@@ -399,6 +439,7 @@ export function useImportResource<D extends object = any>(
     state: {
       loading: state.loading,
```

Comment:
nit: deconstruct these vars
```
const { loading, passwordsNeeded, alreadyExists } = state;
```

Reply:
Here we're returning `state`:

```javascript
return {
  state: {
    loading: state.loading,
    passwordsNeeded: state.passwordsNeeded,
    alreadyExists: state.alreadyExists,
  },
  importResource,
};
```

So I think we could just return:

```javascript
return {
  state,
  importResource,
}
```

But I think we want to return a **copy** of state here, which is why we redefine it ­— I followed the same logic from `useSingleViewResource`:

https://github.com/apache/incubator-superset/blob/4d329071a1651994574c3ad1f26e32b56d9ff810/superset-frontend/src/views/CRUD/hooks.ts#L302-L305

@riahk any thoughts here?

Reply:
Was discussing this with @nytai who originally wrote this logic for `useListViewResource`. I think the original intent was explicitly defining the state variables because some of that hook's state variables are strictly internal. However, in `useSingleViewResource` we're returning all the state variables, so it should be fine to just import the entire `state` object (I will probably refactor the hook to do this at some point).

It looks like that's the case here as well so you can easily just import the entire `state` object like you suggested above.

Reply:
Thanks, @riahk! I've updated the code.